activesupport/lib/active_support/testing/assertions.rb

```diff
@@ -244,6 +244,22 @@ def assert_no_changes(expression, message = nil, from: UNTRACKED, &block)
 
         retval
       end
+
+      private
+        def _assert_nothing_raised_or_warn(&block)
+          assert_nothing_raised(&block)
+        rescue Minitest::UnexpectedError => e
+          if tagged_logger && tagged_logger.warn?
+            warning = <<~MSG
+              #{self.class} - #{name}: #{e.error.class} raised.
+              Assertions nested *inside* `assert_raises` are ignored, this could cause tests to pass incorrectly or fail confusingly.
```

Comment:
Again I'm not good at copy, but `Assertions nested *inside* `assert_raises` are ignored` is weird to me, because they aren't ignored anymore.

Maybe the message should instead say that if the exception was expected, it should be with an `assert_raises` inside the other block assertion?

Reply:
Yeah I'm really struggling getting the wording right here. Thanks for challenging me on it. Pushed a third attempt.